File: api/v1/schemas/strategy.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class StrategyRunRequest(BaseModel):
    code: str
    rsiPeriod: int
    maShortPeriod: int
    maLongPeriod: int
    volumePeriod: int
    overboughtThreshold: float
    oversoldThreshold: float
    stopLossPercent: float
    takeProfitPercent: float

class MetricsBase(BaseModel):
    totalReturn: float = Field(..., allow_inf_nan=False)
    winRate: float = Field(..., allow_inf_nan=False)
    maxDrawdown: float = Field(..., allow_inf_nan=False)
    sharpeRatio: float = Field(..., allow_inf_nan=False)

class StrategyRunResponse(BaseModel):
    code: str
    summary: str
    metrics: MetricsBase
    chartUrl: str

File: apps/dsa-web/src/types/strategy.ts
// ============ 策略Request / Response ============
export interface StrategyParamsRequest {
    code: string;
    rsiPeriod: number;
    maShortPeriod: number;
    maLongPeriod: number;
    volumePeriod: number;
    overboughtThreshold: number;
    oversoldThreshold: number;
    stopLossPercent: number;
    takeProfitPercent: number;
}

export interface Metrics {
    totalReturn: number;
    winRate: number;
    maxDrawdown: number;
    sharpeRatio: number;
}
export interface StrategyResultResponse {
    code:string;
    summary: string;
    metrics: Metrics;
    chartUrl: string; // 假设后端返回图片URL
    bestStrategyDescription:string;
}
